packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -1,128 +1,747 @@
+    delete res.lotNumber;
+
+    if (res.vaccineCode) {
+      const resVaccineCodeString = JSON.stringify(res.vaccineCode).toLowerCase();
+      if (
+        resVaccineCodeString.includes("no data") ||
+        resVaccineCodeString.includes("no immunization")
+      ) {
+        return undefined;
+      }
+
+      updRes.vaccineCode = getUniqueDisplays(res.vaccineCode);
+    }
+
+    delete res.doseQuantity;
+  }
+
+  if (res.resourceType === "Practitioner") {
+    const name = getNameString(res.name);
+    updRes.name = name;
+
+    const qualificationsText = getUniqueDisplays(res.qualification?.[0]?.code);
+    if (qualificationsText) updRes.qualification = qualificationsText;
+
+    updRes.address = getAddressString(res.address);
+  }
 
-  if (!bundle.entry?.length) return undefined;
-  const dedupedBundle = deduplicateBundleResources(bundleCopy);
-  if (!dedupedBundle.entry?.length) return undefined;
-  const filteredString = filterBundleResources(dedupedBundle);
-  if (!filteredString) return undefined;
-  return (await preprocessDataWithRag(filteredString)).trim();
+  if (res.resourceType === "Procedure") {
+    const name = getUniqueDisplays(res.code);
+    if (name) {
+      updRes.name = name;
+      if (name.includes("no data")) return undefined;
+    }
+
+    delete updRes.code;
+
+    if (updRes.status === "") delete updRes.status;
+
+    delete res.reasonCode; // TODO: Introduce term server lookup here
+  }
+
+  if (res.resourceType === "DiagnosticReport") {
+    const code = res.code?.text;
+    if (code) updRes.code = code;
+
+    const category = res.category
+      ?.map(cat => cat.coding?.flatMap(coding => coding.display || []))
+      .join(", ");
+    if (category) updRes.category = category;
+
+    if (res.presentedForm) {
+      updRes.presentedForm = res.presentedForm.map(form => {
+        delete form.contentType;
+        if (form.data) {
+          form.data = Buffer.from(form.data, "base64").toString("utf-8");
+        }
+        return form;
+      });
+    }
+  }
+
+  if (res.resourceType === "Observation") {
+    if (res.category) {
+      const category = getUniqueDisplays(res.category);
+      if (category) updRes.category = category;
+    }
+
+    const code = getUniqueDisplays(res.code);
+    if (code) updRes.reading = code;
+
+    if (res.valueCodeableConcept) {
+      updRes.value = getUniqueDisplays(res.valueCodeableConcept);
+      delete updRes.valueCodeableConcept;
+    }
+
+    if (res.valueQuantity) {
+      updRes.value = getQuantityString(res.valueQuantity);
+      delete updRes.valueQuantity;
+    }
+
+    if (res.interpretation) {
+      updRes.interpretation = getUniqueDisplays(res.interpretation);
+    }
+
+    if (res.referenceRange) {
+      updRes.referenceRange = res.referenceRange.map(range => {
+        const low = getQuantityString(range.low);
+        const high = getQuantityString(range.high);
+        return {
+          low,
+          high,
+        };
+      });
+    }
```

Comment:
nit: a summary of what this is doing would be really nice, and/or break this into specific functions per-resrouce

a single 150 LOC function with no comments will make a dev want to blow their brains out from a maintainability perspective

Reply:
added a comment for now, but will also add TODO to split this up